ensure_tiller_installed.go

```diff
@@ -261,10 +261,10 @@ func (c *Client) EnsureTillerInstalledWithValues(ctx context.Context, values []s
 
 		o := func() error {
 			t, err := c.newTunnel()
-			if !installTiller && IsTillerNotFound(err) {
-				return backoff.Permanent(microerror.Mask(err))
-			} else if err != nil {
+			if IsTillerNotFound(err) || IsTooManyResults(err) || IsTillerInvalidVersion(err) {
 				return microerror.Mask(err)
+			} else if err != nil {
+				return backoff.Permanent(microerror.Maskf(tillerStartingError, "can't establish a tunnel through new tiller pod with error %#q", err))
```

Comment:
```suggestion
				return backoff.Permanent(microerror.Maskf(tillerStartingError, "can't establish  tunnel to tiller pod with error %#q", err))
```

So you're going to match on this error in the tiller resources?

Reply:
@rossf7 Yes, I will use it on tiller resource and would cancel reconciliation.